scipy/_lib/_array_api.py

```diff
@@ -600,140 +601,303 @@ def is_marray(xp):
+        combinations of backend and device (or other capability) are supported.
+
+        {textwrap.indent(table, ' ' * 8).lstrip()}
+
+        See :ref:`dev-arrayapi` for more information.
+        """
+        note = textwrap.dedent(note)
+
+        doc = FunctionDoc(self.func)
+        doc['Notes'].append(note)
+        self.func.__doc__ = str(doc).split("\n", 1)[1]  # remove signature
 
+    def _lazy_xp_function(self):
+        """Tag self.func with xpx.lazy_xp_function."""
+        from scipy._lib.array_api_extra.testing import lazy_xp_function
 
-def xp_capabilities(name=None, capabilities_table=None, **kwargs):
-    capabilities_table = (xp_capabilities_table if capabilities_table is None
-                          else capabilities_table)
+        assert self.func
+        lazy_xp_function(self.func, jax_jit=self.jax.jit,
+                         static_argnames=self.jax.static_argnames,
+                         static_argnums=self.jax.static_argnums,
+                         allow_dask_compute=self.dask.allow_compute)
 
-    def decorator(f):
-        fun_name = f.__name__
-        table_entry = name or fun_name
-        if table_entry not in capabilities_table:
-            capabilities_table[table_entry] = kwargs
-        capabilities = _make_capabilities(**capabilities_table[table_entry])
+    def _make_pytest_marks(self):
+        import pytest
 
-        @functools.wraps(f)
-        def wrapper(*args, **kwargs):
-            return f(*args, **kwargs)
+        return [
+            getattr(pytest.mark, mark)(*args, **kwargs)
+            for mark, args, kwargs in self._pytest_marks
+        ]
 
-        note = _make_capabilities_note(fun_name, capabilities)
-        doc = FunctionDoc(wrapper)
-        doc['Notes'].append(note)
-        wrapper.__doc__ = str(doc).split("\n", 1)[1]  # remove signature
+    def test_case(self, test_func):
+        """Return a pytest decorator and apply lazy_xp_function::
 
-        return wrapper
-    return decorator
+            from scipy.mymod import func
+
+            @func.capabilities.skip_xp_backends
+            def test_func(..., xp):
+                ...
+
+        expands to::
 
+            from scipy.mymod import func
+            from scipy._lib.array_api_extra.testing import lazy_xp_function
 
-def make_skip_xp_backends(fun_name, capabilities_table=None):
-    capabilities_table = (xp_capabilities_table if capabilities_table is None
-                          else capabilities_table)
+            lazy_xp_function(func, ...)
+            
+            @pytest.mark.skip_xp_backends(...)
+            @pytest.mark.skip_xp_backends(...)
+            @pytest.mark.skip_xp_backends(...)
+            def test_func(xp):
+                ...
+        """
+        self._lazy_xp_function()
+        for mark in self._make_pytest_marks():
+            test_func = mark(test_func)
+        return test_func
 
-    import pytest
+    def pytest_param(self):
```

Comment:
This is unused for now but it will feature prominently in `scipy.special.tests.test_alternative_backends`, and in general whenever functions are passed to `@pytest.mark.parametrize`